homeassistant/components/tplink/entity.py

```diff
@@ -210,7 +229,13 @@ def _entities_for_device[_E: CoordinatedTPLinkEntity](
     for device features that are implemented by specialized platforms like light.
     """
     return [
-        entity_class(device, coordinator, feature=feat, parent=parent)
+        entity_class(
+            device,
+            coordinator,
+            feature=feat,
+            parent=parent,
+            add_to_parent=feat.category == Feature.Category.Primary,
```

Comment:
Do we really need a separate flag given that we could perform the check inside the ctor?

Reply:
The flag is needed for the sensor platform because it's currently created without a feature being passed but it's "non-primary" and there's no way to tell inside the `entity` constructor without introspecting `__class__`.  I've removed it for now and the emeter sensors are now on the parent but with https://github.com/home-assistant/core/pull/117764 they should go back on the children.